pandas-stubs/core/strings.pyi

```diff
@@ -5,17 +5,25 @@ from typing import (
     Generic,
     Literal,
     Sequence,
+    TypeVar,
     overload,
 )
 
 import numpy as np
 import pandas as pd
-from pandas import Series
+from pandas import (
+    DataFrame,
+    MultiIndex,
+    Series,
+)
 from pandas.core.base import NoNewAttributesMixin
 
 from pandas._typing import T
 
-class StringMethods(NoNewAttributesMixin, Generic[T]):
+# The _TS type is what is used for the result of str.split with expand=True
+_TS = TypeVar("_TS", DataFrame, MultiIndex)
+
+class StringMethods(NoNewAttributesMixin, Generic[T, _TS]):
```

Comment:
how is the value of `_TS` determined? It is not in `__init__` and you didn't add it explicitly (return value of `StringMethods[MultiIndex]`)

Reply:
It is passed in as a parameter to the `StringMethods` class so we know what the return type should be of `str.split` when `expand==True`

Reply:
sorry, I'm blind - I didn't see the annotation of `str()` :)